modules/ROOT/pages/_partials/partner-and-customer-guide.adoc

```diff
@@ -0,0 +1,158 @@
+
+// tag::before-you-begin[]
+
+== Before You Begin
+
+Before upgrading and testing your custom connector for Java 17, you must be familiar with the following considerations:
+
+* Target compilation level
++
+Ensure your custom connector runs with Java 17, but always use Java 8 as the target compilation level.
++
+Because you must compile your custom connector with Java 8, use the Java compiler configuration from the Java SDK parent POM. You can’t use language features from Java 11 or Java 17. 
+
+* Unsupported directives
++
+Do not set `-add-opens` or `-add-export` directives.
+
+
+// end::before-you-begin[]
+
+// tag::mule-sdk-connectors[]
+
+[[upgrade-your-mule-sdk-connectors]]
+== Upgrade Your Custom Mule SDK Connectors
+
+Upgrade your custom Mule SDK (Java SDK and XML SDK) connectors to Java 17.
+
+[[upgrade-third-party-libraries]]
+=== Upgrade Third-Party Libraries
+
+Upgrade the third-party libraries of your custom connector to be compatible with Java 17:
+
+* Java EE libraries
++
+Mule 4.6 exposes Java EE libraries differently than earlier Mule versions. You must add the BOM dependency to your custom connector and, if applicable, exclude the provided conflicting library. You can also add the libraries in the BOM dependency separately.
++
+[source,java,linenums]
+----
+<muleJavaEeBomVersion>4.6.0-20231025</muleJavaEeBomVersion>
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>org.mule</groupId>
+        <artifactId>mule-javaee-runtime-bom</artifactId>
+        <version>${muleJavaEeBomVersion}</version>
+        <type>pom</type>
+        <scope>import</scope>
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
+----
+
+* External libraries
++
+Upgrade external libraries, which are third-party libraries that are not bundled with your custom connector. Some examples include a JDBC driver, JMS broker client, or Groovy runtime. The following example shows a JDBC driver external library configuration: 
++
+[source,java,linenums]
+----
+@ExternalLib(name = "MySQL JDBC Driver",
+description = "A JDBC driver that supports connection to the MySQL Database",
+nameRegexpMatcher = "(.*)\\.jar",
+requiredClassName = "com.mysql.jdbc.Driver",
+coordinates = "mysql:mysql-connector-java:5.1.44")
+public class MySqlConnectionProvider implements ConnectionProvider<Connection> {
+  //
+}
+----
++
+You must update your external libraries to a version that is compatible with Java 17. 
+
+=== Review Your Code
+
+After you compile your custom connector with Java 8, ensure you can run your custom connector with Java 17. Reviewing your code is important because some restrictions are flagged as warnings in Java 11 but are flagged as failures and can’t be disabled in Java 17. For example, reflection is more restricted in Java 17, so you must review your code to ensure there are no illegal actions. 
+
+To help you review your code, learn more about the different changes in Java 17:
+
+* JDK Migration Guide
```

Comment:
Can we double confirm if we need to check with the Oracle guide or Open JDK distribution .